### PR TITLE
Likes for tips and stories

### PR DESCRIPTION
**Pull request by katbow**
Users can add likes for all posts types anywhere they are displayed.

### CODE REVIEW

**Comment by katbow** [comment]
@iteles @nelsonic  - I didn't add in 'unliking' or disabling likes at the moment - I focussed on giving feedback to the user after they like something for now & capturing likes (regardless if they are duplicate by a user)

File: web/controllers/post_controller.ex [hunk 1]
```diff
@@ -34,6 +34,22 @@ plug :authenticate when action in [:new, :create]
     render(conn, "index.html", posts: posts)
   end
 
+  def likes(conn, %{"id" => id}) do
```
Comment (tonydaly) [inline web/controllers/post_controller.ex]: In an aim to be more RESTful I kinda feel like this should actually be the `#create` action in a `LikesController`? But we could keep the current URL structure the same, we should be able to do this with nested routes.

File: web/templates/post/index.html.eex [hunk 2]
```diff
@@ -5,6 +5,25 @@
       <%= markdown String.slice(post.content, 0, 325) <> "..." %>
     </p>
     <a href="/posts/<%= post.id %>" class="pv2 ph2 f5 f4-l lh-copy">Read this story</a>
+    <%= if @current_user do %>
```
Comment (tonydaly) [inline web/templates/post/index.html.eex]: Seems pretty similar to what's in `web/templates/tip/index.html.eex` is there an easy way to DRY it up?

File: web/templates/tip/index.html.eex [hunk 3]
```diff
@@ -3,14 +3,33 @@
...
                 String.contains?(post.content, "#connect") -> "connect"
                 String.contains?(post.content, "#keeplearning") -> "keeplearning"
                 String.contains?(post.content, "#givetoothers") -> "givetoothers"
                 String.contains?(post.content, "#beactive") -> "beactive"
                 String.contains?(post.content, "#takenotice") -> "takenotice"
                 true -> "no tag found"
               end%>">See more "<%= tag %>" tips</a>
+      <%= if @current_user do %>
+        <%= user_likes = @current_user.likes |> Enum.map(fn(p) -> p.id end) %>
+        <%= if Enum.member?(user_likes, post.id) do %>
```
Comment (tonydaly) [inline web/templates/tip/index.html.eex]: Did we figure out how to do this in one query?

File: web/templates/page/index.html.eex [hunk 4]
```diff
@@ -17,14 +36,33 @@
   <div class="pa3 bg-near-white b--black-10">
     <%= if @tip do %>
       <p class="pv2 ph2 ph3-m ph4-l f5 f4-l lh-copy"><%= markdown @tip.content %></p>
-      <a href="/tips/?tag=<%= tag = cond do
+      <a href="/tips?tag=<%= tag = cond do
```
Comment (tonydaly) [inline web/templates/page/index.html.eex]: This feels strange 🤔

Ideally, we'd like as little code/logic in our templates as possible, could this be moved into a helper in the `View`?

**Comment by tonydaly** [comment]
Nice one @katbow!

**Comment by nelsonic** [comment]
@katbow can you please convert @tonydaly's comments into a `technical` issue with a checklist?
I _agree_ that this is not a "_blocker_" so think we can _merge_ it today... 👍

**Pull request merged**